lib/improver/cli/nbhood.py

```diff
@@ -372,27 +164,26 @@ def process(cube, neighbourhood_output, neighbourhood_shape, radius=None,
         raise RuntimeError('mask_cube cannot be used with'
                            'neighbourhood_output="circular"')
 
-    if neighbourhood_shape == 'circular' and apply_recursive_filter:
-        raise ValueError('Recursive filter option is not applicable to '
-                         'circular neighbourhoods. ')
-
     if degrees_as_complex:
         if neighbourhood_output == "percentiles":
             raise ValueError(
                 'Cannot generate percentiles from complex numbers')
         if neighbourhood_shape == "circular":
             raise ValueError(
                 'Cannot process complex numbers with circular neighbourhoods')
-        if apply_recursive_filter:
-            raise ValueError(
-                'Cannot process complex numbers with recursive filter')
 
     if degrees_as_complex:
         # convert cube data into complex numbers
         cube.data = WindDirection.deg_to_complex(cube.data)
 
-    radius_or_radii, lead_times = radius_or_radii_and_lead(
-        radius, radii_by_lead_time)
+    if lead_times is None:
+        radius_or_radii = float(radii[0])
+    else:
+        if len(radii) != len(lead_times):
+            raise RuntimeError("If leadtimes are supplied, it must be a list"
+                               " of equal length to a list of radii.")
+        radius_or_radii = [float(x) for x in radii]
+        lead_times = [int(x) for x in lead_times]
```

Comment:
I assume this is doing something slightly different to the original function but it's now used in at least two neighbourhooding CLIs.  Can this be a utility somewhere?

Reply:
I agree. I am intending to move it in the nbhood-iterate-with-mask PR but I can't decide where it should live. Any thoughts?

Reply:
NB cli_utilities is intended for removal by the green team in my understanding.

Reply:
OK, in that case I think this is a general neighbourhood processing utility: so it could maybe live in `improver.nbhood` somewhere.  There is no general utilities file in here, so you could create one, put it in `__init__` or choose one of the other files to put it in for now.